src/rust/pipeline-ingress/tests/integration_test.rs

```diff
@@ -0,0 +1,186 @@
+            endpoint = %endpoint,
+        );
+
+        HealthcheckClient::wait_until_healthy(
+            endpoint.clone(),
+            "graplinc.grapl.api.pipeline_ingress.v1beta1.PipelineIngressService",
+            Duration::from_millis(10000),
+            Duration::from_millis(500),
+        )
+        .await
+        .expect("pipeline-ingress never reported healthy");
+
+        tracing::info!("connecting pipeline-ingress gRPC client");
+        let grpc_client = PipelineIngressClient::connect(endpoint.clone())
+            .await
+            .expect("could not configure gRPC client");
+
+        PipelineIngressTestContext {
+            grpc_client,
+            bootstrap_servers,
+            sasl_username,
+            sasl_password,
+            consumer_group_name,
+            _guard,
+        }
+    }
+}
+
+#[test_context(PipelineIngressTestContext)]
+#[tokio::test]
+async fn test_publish_raw_log_sends_message_to_kafka(ctx: &mut PipelineIngressTestContext) {
+    let event_source_id = Uuid::new_v4();
+    let tenant_id = Uuid::new_v4();
+    let log_event: Bytes = "test".into();
+
+    let bootstrap_servers = ctx.bootstrap_servers.clone();
+    let sasl_username = ctx.sasl_username.clone();
+    let sasl_password = ctx.sasl_password.clone();
+    let consumer_group_name = ctx.consumer_group_name.clone();
+
+    tracing::info!("configuring kafka consumer");
+    let kafka_consumer = Consumer::new(
+        bootstrap_servers,
+        sasl_username,
+        sasl_password,
+        consumer_group_name,
+        "raw-logs".to_string(),
+    )
+    .expect("could not configure kafka consumer");
+
+    tracing::info!("creating kafka subscriber thread");
+    let kafka_subscriber = tokio::task::spawn(async move {
+        let contains_expected = kafka_consumer
+            .stream()
+            .expect("could not subscribe to the raw-logs topic")
+            .any(|res: Result<Envelope<RawLog>, ConsumerError>| async move {
+                let envelope = res.expect("error consuming message from kafka");
+                let metadata = envelope.metadata;
+                let raw_log = envelope.inner_message;
+                let expected_log_event: Bytes = "test".into();
+
+                tracing::debug!(message = "consumed kafka message");
+
+                metadata.tenant_id == tenant_id
+                    && metadata.event_source_id == event_source_id
+                    && raw_log.log_event == expected_log_event
+            });
+
+        tracing::info!("consuming kafka messages for 30s");
+        assert!(
+            tokio::time::timeout(Duration::from_millis(30000), contains_expected)
+                .await
+                .expect("failed to consume expected message within 30s")
+        );
+    });
+
+    // we throw in a healthy sleep here to make well sure the kafka consumer is
+    // actually consuming before we throw a message at it
+    tokio::time::sleep(Duration::from_millis(5000)).await;
```

Comment:
5 seconds seems really high, shouldn't it pretty much be immediately listening for messages?

Reply:
It is high locally. But I'd like to give this a good chance of succeeding in a cloud environment as well, and until we have reliable metrics I can't know what a reasonable timeout is. So I've set this (and the others) conservatively quite high with the hope that this will be a reliable test in the `testing` pipeline (e.g. AWS), and we can turn it down once we have a higher fidelity understanding of the actual environment.